### ISSUE TITLE
IntelliJ immediate crash when opening any bazel project

### ISSUE THREAD

**Issue by qq7te** [bazelbuild/intellij#3852]
I'm installing IDEA  on ubuntu 22.04 using the toolbox.
No matter the version of intelliJ IDEA (2021.2, 2021.3, 2022.1, 2022.2) with the bazel plugin (I've tried the latest three versions already, including 2022.08.11.0.1-api-version-222), I constantly get a failure as I open any python project, no matter how small: 
```

java.util.concurrent.ExecutionException: com.google.idea.blaze.base.command.info.BlazeInfoException: blaze info failed with exit code: -1
	at com.google.common.util.concurrent.AbstractFuture.getDoneValue(AbstractFuture.java:588)
	at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:567)
	at com.google.common.util.concurrent.FluentFuture$TrustedFuture.get(FluentFuture.java:91)
	at com.google.idea.blaze.base.async.FutureUtil$Builder.lambda$run$0(FutureUtil.java:109)
	at com.google.idea.blaze.base.scope.Scope.push(Scope.java:40)
	at com.google.idea.blaze.base.async.FutureUtil$Builder.run(FutureUtil.java:99)
	at com.google.idea.blaze.base.sync.ProjectStateSyncTask.getProjectState(ProjectStateSyncTask.java:136)
	at com.google.idea.blaze.base.sync.ProjectStateSyncTask.collectProjectState(ProjectStateSyncTask.java:66)
	at com.google.idea.blaze.base.sync.BlazeSyncManager.lambda$requestProjectSync$1(BlazeSyncManager.java:129)
	at com.google.idea.blaze.base.scope.Scope.push(Scope.java:57)
	at com.google.idea.blaze.base.scope.Scope.root(Scope.java:33)
	at com.google.idea.blaze.base.sync.BlazeSyncManager.lambda$requestProjectSync$2(BlazeSyncManager.java:111)
	at com.google.idea.blaze.base.async.executor.ProgressiveTaskWithProgressIndicator.lambda$submitTask$0(ProgressiveTaskWithProgressIndicator.java:83)
	at com.google.idea.blaze.base.async.executor.ProgressiveTaskWithProgressIndicator.lambda$submitTaskWithResult$4(ProgressiveTaskWithProgressIndicator.java:131)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:66)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:188)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:608)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:683)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:639)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:607)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:175)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:66)
	at com.google.idea.blaze.base.async.executor.ProgressiveTaskWithProgressIndicator.lambda$submitTaskWithResult$5(ProgressiveTaskWithProgressIndicator.java:131)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:131)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:74)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:82)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:241)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:31)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.execute(BoundedTaskExecutor.java:214)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:212)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:203)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: com.google.idea.blaze.base.command.info.BlazeInfoException: blaze info failed with exit code: -1
	at com.google.idea.blaze.base.command.info.BlazeInfoRunnerImpl.runBlazeInfo(BlazeInfoRunnerImpl.java:102)
	at com.google.idea.blaze.base.command.info.BlazeInfoRunnerImpl.lambda$runBlazeInfo$2(BlazeInfoRunnerImpl.java:72)
	... 15 more

```

I can't possibly be the only one... What am I doing wrong?

**Comment by qq7te** [comment]
To add to it, I see the same failure in CLion 2022.2 as well !

**Comment by qq7te** [comment]
Absolutely, will do momentarily. Just noticed that this happens only with IntelliJ and Clion installed by the Toolbox, so there's another interesting data point for you.

**Comment by tpasternak** [comment]
@qq7te I'm not sure if this is the case, but I'm getting this when I forget to setup bazel executable in Settings -> Bazel Settings

**Comment by qq7te** [comment]
I'm really lagging behind here. My apologies. I think you're probably right with your suggestion, I need a moment to test these... please don't close my ticket as abandoned yet... :grin:

**Comment by qq7te** [comment]
I have some updates:
This happens because I am using `bazelisk` . Instead of having a proper `bazel` installed, my `bazel` executable is really a script that selects the right version of bazel for my project and downloads and runs it. 

In order to run `bazelisk` , `node` needs to be in the path, since `bazelisk` is a javascript tool.

In my case, `node` is in the `PATH` when I use the terminal, but not when I start bazel within IntelliJ.
I have tried to modify my `.bashrc` so that it executes it all even in non-interactive mode but that doesn't seem to make a difference. 

The problem is basically that all I have in the PATH is set before `.bashrc` is executed and even if I have put the proper directories in the PATH as an environment setting in the IntelliJ   (at "Appearance & Behavior" > "Path Variables") , it still doesn't seem to work.

**So, this is not a problem of bazel plugin .... it's really a strange problem with not being able to set the path appropriately.**

I have a workaround: I know that `~/.local/bin` is in the `PATH` at the beginning so I just symlinked `node` in there and everything works well.

Unless I'm doing something strange, this is probably going to happen to anyone who is using `bazelisk` to manage `bazel` versions in their projects. What can we do to solve it? 

To reproduce, in a relatively clean system:
* don't install bazel
* install bazelisk as an npm module
* make sure your project has a `.bazelversion` file which allows `bazelisk` to pick the right version of `bazel`
* Open your project with  IntelliJ with bazel plugin
* ..... error ....

Let me know if this is helpful
